### ISSUE TITLE
ZipArchive signals SubscriptOutOfBounds while extracting valid archive

### ISSUE THREAD

**Issue by Rinzwind** [guillep/pharo-core#14235]
**Bug description**
ZipArchive signals a SubscriptOutOfBounds while extracting the following archive: [test.zip](https://github.com/pharo-project/pharo/files/11971321/test.zip)

I assume the archive is valid as the ‘unzip’ command in macOS 12.6.7 detects no errors when testing it:

```
$ unzip -t test.zip 
Archive:  test.zip
    testing: directory/               OK
    testing: directory/file2          OK
    testing: directory/file1          OK
No errors detected in compressed data of test.zip.
```

But the following signals a SubscriptOutOfBounds within the context of the `#extractAllTo:` message:

```smalltalk
ZipArchive new
	readFrom: FileLocator imageDirectory / 'test.zip';
	extractAllTo: FileLocator temp / Time microsecondClockValue asString
```

**Version information:**
Pharo-12.0.0+build.558.sha.eabccb3bcafe951223e983e270a735d8774e47e2 (64 Bit)

**Comment by Rinzwind** [comment]
I thought this would be due to a bug in the decompression algorithm, but the workaround applied in pull request #16806 points to a bug in ZnBufferedReadStream instead, so I traced the messages sent to the stream to check for a difference between ReadStream and ZnBufferedReadStream, which led me to the bug I described in [Zinc issue #148](https://github.com/svenvc/zinc/issues/148).

**Comment by Rinzwind** [comment]
I'll close this as it was fixed through pull request #17055.
